pkg/render/logstorage/esmetrics/elasticsearch_metrics.go

```diff
@@ -89,6 +97,13 @@ func (e *elasticsearchMetrics) Objects() (objsToCreate, objsToDelete []client.Ob
 	toCreate = append(toCreate, secret.ToRuntimeObjects(secret.CopyToNamespace(render.ElasticsearchNamespace, e.cfg.ESMetricsCredsSecret)...)...)
 	toCreate = append(toCreate, e.metricsService(), e.metricsDeployment(), e.serviceAccount())
 
+	if e.cfg.UsePSP {
+		toCreate = append(toCreate,
+			e.metricsRole(),
```

Comment:
should the role and binding be removed if !usePSP?

Reply:
I had a similar discussion with one of our operator developers. We are not deleting PSP related resources (policy itself, (Cluster)Role, and (Cluster)RoleBinding) when `usePSP` is `false`. Maybe we can address this together in a future PR.